policyengine_us/variables/gov/states/ny/tax/income/taxable_income/deductions/itemized/ny_itemized_deductions_reduction.py

```diff
@@ -7,23 +7,83 @@ class ny_itemized_deductions_reduction(Variable):
     label = "NY itemized deductions reduction"
     unit = USD
     definition_period = YEAR
-    reference = "https://www.nysenate.gov/legislation/laws/TAX/615"
+    reference = "https://www.nysenate.gov/legislation/laws/TAX/615"  # (f)&(g)
     defined_for = StateCode.NY
 
     def formula(tax_unit, period, parameters):
         p = parameters(
             period
-        ).gov.states.ny.tax.income.deductions.itemized
+        ).gov.states.ny.tax.income.deductions.itemized.reduction
         agi = tax_unit("ny_agi", period)
+        filing_status = tax_unit("filing_status", period)
         itemized_deduction = tax_unit("ny_itemized_deductions_max", period)
-        #if agi <= 475_000; single: 100_000, joint: 20_000, hoh: 250_000
-        #    return 0
-        #if 475_000 < agi <= 525_000;
-        #    return 25% * itemized_deduction
-        #else: 
-        #    return 50% * itemized_deduction
-        return p.amount.calc(agi) * itemized_deduction
-        # if 1_000_000 < agi <= 10_000_000, 
-            return 50% * charitaele_contributions
-        # if 10_000_000 < agi, 
-            return 25% * charitaele_contributions
\ No newline at end of file
+        charitable_deduction = tax_unit("charitable_deduction", period)
+        itemized_deduction_less_charity = (
+            itemized_deduction - charitable_deduction
+        )
+        # Income threshold for non-charitable deductions reduction
+        first_reduction_threshold = p.amount.start[filing_status]  # 100_000
+        second_reduction_threshold = p.amount.end  # 475_000
+        # Income threshold where only charitable deduction applies
+        high_income_threshold = p.amount.high_income
+        higher_income_threshold = p.amount.higher_income
+
+        no_adjustment_condition = agi <= first_reduction_threshold
```

Comment:
Lets make this a separate `ny_itemized_deductions_reduction_applies.py` variable and define the reduction for that condition